ImageCropper/Cropper.py

```diff
@@ -0,0 +1,169 @@
+# Image Cropper
+# 2023.03.26 ~ 28
+
+
+# Libraries
+from PIL import Image                                       # PIL; Python Imaging Library
+import json                                                 # json.load(); call coordinates information for cropping from the external JSON file
+import os                                                   # os.path.*, os.getcwd(), os.makedirs(); deal with paths in various ways
+import sys                                                  # sys.argv; read "test" arguement from the terminal command or batchfile
+import pprint                                               # pprint.pprint(); print multi-line data like JSON with line replacement
+
+
+# Global variables
+test = False
+
+
+# Find if test mode
+def FindIfTest():
+
+    global test                                             # must be declared in each udf
```

Comment:
모든 함수에 global test를 일일이 추가하는 것 보다, wrapper function을 만드는 것이 깔끔할 수도 있어요.
```
test = False

def FindIfTest():
    global test
    ....

def IsTest():
    global test
    return test

def ReadJson():
    ....
    if IsTest():
        ...
```

이렇게 하면, 각 함수에서 `global test`를 안써도 되는 이점이...

Reply:
외부 코드에서 함수 위에 쓰인 데코레이터를 보고 이걸 왜 쓰나 했었는데, 지금 상황에 비추어 다시 개념을 찾아보니 어떤 이점이 있는지 비로소 이해가 된 것 같습니다. 감사합니다ㅎ

Reply:
제가 생각한 그 래퍼 함수를 말씀하신 게 아니군요. 첫 리뷰부터 잘못 이해해서 죄송합니다;;

지금 다시 이해한 게 맞다면, 아래 `os.getenv()`를 쓰면 한 번의 global 선언조차 없을 것 같네요. 비용에서 큰 이슈가 있을 것 같지도 않구요(이런 비용을 따질 만한 코드가 아닙니다만ㅋㅋ).

Reply:
>> os.getenv()를 쓰면 한 번의 global 선언조차 없을 것 같네요. 비용에서 큰 이슈가 있을 것 같지도 않구요

혹시 os.getenv()를 매번 각 함수에서 매번 체크를 하는 것을 의미하셨나요? 마치 아래의 코드처럼요.
```
def FindIfTest():
    """returns true if TEST is set with 1. False, otherwise.
   
       e.g., $ TEST=1 python3 Cropper.py
    """
    return os.getenv("TEST", "0")  == "1"
```

그렇게도 괜찮을 것 같기는 한데, 정확히 말하자면, global variable을 사용하는 코드를 줄이는 의미로써,  wrapper function을 추천드린 것 이었어요.

````
is_test = False

def FindIfTest():
    return is_test

def Fun1(...):
    if FindIfTest():
        ...
   ....
   if FindIfTest():
       ...
    

def Func2(...):
    if FindIfTest():

```

위의 코드에서 `is_test`라는 글로벌 변수를 직접 access하는 FindIfTest() 뿐이죠. 코드가 길어지고 복잡해지면, 의도치 않게 global 변수와 같은 이름의 로컬 변수를 선언하는 일도 발생하고, 의도치 않게 글로벌 변수에 저장된 값을 변경시킬 수도 있어요. 특히나 그것마저 특정 조건에서만 일어날 수도 있고요. 그래서 글로벌 변수를 사용하는 것은 code maintenance에 너무 안좋아서 pylint도 쓰지말라고 하고, 대다수의 프로그래밍 언어들에서 왠만하면 글로벌 변수를 피해보라고 권장을 하죠.

Reply:
제 설명이 불필요하게 길었던 것 같아서, 부연설명을 하자면, 글로벌 변수의 직접 접근을 제한하고 최소화하기 위해, FindIfTest()같이 wrapper function에서만 접근하게 하고, 비지니스 로직들에선 그런 함수들을 통해서만 접근하는 것이 나을 것 같다는 저의 의견이었습니다.

Reply:
네 맨 위에서 하신 말씀을 `test`를 글로벌로 한 번은 선언해서 쓰되 이후에 접근을 IsTest()를 통해서만 하자는 것으로 이해했고, 여기서 기왕에 글로벌 변수 사용을 줄이고자 한다면 아예 안 쓰는 한 방법으로서, 다시 말씀해주신 것과 같은 방식으로 다른 함수 안에서 `FindIfTest()`가 실행될 때마다 그 안에 있는 `os.getenv()`도 재실행되는 구조를 생각해봤습니다. 프로그램 개시 시점부터 변하지 않는 값을 매번 새로 함수를 통해 불러오는 게 좀 어색하지만, 실질적인 내용이 환경변수 안에 저장되는 속성값 하나를 탐색해서 불러오는 정도라면 비용면에서 괜찮지 않은가 하는 관점으로 ……

하지만 지금 해주신 말씀은 너무 무리해서까지 글로벌 변수를 박멸(?)하진 말고 중간 타협점에 대한 이야기이신 것 같습니다. 저도 지금 생각은 글로벌 변수를 완전히 없애기 위한 수고가 너무 커진다면(가령 지금껏 만든 모든 함수를 클래스 안의 메서드로 집어넣고 일일이 `self.`를 추가한다든가. 이것만은 하고 싶지 않습니다ㅋㅋㅋ) 그냥 글로벌 변수를 조금 더 조심해서 쓰자는 정도입니다.